### ISSUE TITLE
Clone failure with buf 0.47.0

### ISSUE THREAD

**Issue by charithe** [bufbuild/buf-breaking-action#10]
I started seeing this failure from `buf-breaking-action` after the release of 0.47.0:

```
Failure: could not clone https://github.com/myorg/myrepo.git: exit status 129
error: unknown option `config'
usage: git fetch [<options>] [<repository> [<refspec>...]]
...
```

My suspicion is that it is due to the new git clone implementation in 0.47.0.

**Comment by mfridman** [comment]
Thanks for reporting this @charithe, we'll take a look.

**Comment by mfridman** [comment]
We've identified the issue and pushes a fix: https://github.com/bufbuild/buf/releases/tag/v0.48.1.

If you're using the `buf` setup action (without a version), it should pull latest, e.g.,

```
- uses: bufbuild/buf-setup-action@v0.3.1
```

Otherwise update to using:`v0.48.1`  

@charithe Please let us know if you have further issues.

**Comment by charithe** [comment]
Thanks @mfridman that was quick!

Unfortunately, now it seems to be failing due to git LFS.

```
Failure: could not clone https://github.com/myorg/myrepo.git: exit status 128
Error: Failure: could not clone https://github.com/myorg/myrepo.git: exit status 128
fatal: could not read Username for 'https://github.com': No such device or address
Downloading docs/modules/ROOT/assets/images/how_it_works.png (64 KB)
fatal: could not read Username for 'https://github.com': No such device or address
Error downloading object: docs/modules/ROOT/assets/images/how_it_works.png (c8889f0): Smudge error: Error downloading docs/modules/ROOT/assets/images/how_it_works.png (c8889f0a9786553413d49edbbab93739de2988e8b9f8f33df8d1a7368453e715): batch response: Git credentials for https://github.com/myorg/myrepo.git not found.

Errors logged to /.git/lfs/logs/20210730T162028.685205191.log
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: docs/modules/ROOT/assets/images/how_it_works.png: smudge filter lfs failed.
```

**Comment by mfridman** [comment]
Thanks for reporting this, we didn't test with LFS but it's good to know this use-case. 

We've pushed another fix: `v0.48.12`. For more info see https://github.com/bufbuild/buf/releases/tag/v0.48.2

Give this another try and if issues persist let us know.

cc @charithe

**Comment by charithe** [comment]
That has fixed the issue. Thank you for the quick response.
